tools/whitelist/WorkspaceDepsWhiteListTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.whitelist;
+
+import org.junit.Test;
+import org.junit.Before;
+import java.util.Map;
+import java.util.Set;
+import java.util.List;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.FileInputStream;
+import java.io.File;
+
+import static org.junit.Assert.assertTrue;
+
+import org.yaml.snakeyaml.Yaml;
+
+public class WorkspaceDepsWhiteListTest {
+
+  private static class WorkspaceElement {
+    private String label;
+    public Map<String, String> attributes;
+
+    public WorkspaceElement(String label) {
+      this.label = label;
+      this.attributes = new HashMap<String, String>();
+    }
+
+    public void setAttribute(String key, String value) {
+      this.attributes.put(key, value);
+    }
+
+    public String getLabel() {
+      return this.label;
+    }
+
+    public String getAttribute(String key) {
+      return this.attributes.get(key);
+    }
+  }
+
+  private List<WorkspaceElement> parseLines(List<String> lines) {
```

Comment:
Let me suggest a different approach:
For each line, do:
* Split into "=" and fail if not a proper key-value
* If key is `url`, `urls`, `remote`, `artifact` etc., validate value.
      Else, if value contains `://`, fail (this would mean we need to add a key to the list)